### DISCUSSION TITLE
Debugging failed builds because of missing layer

### DISCUSSION THREAD

**Discussion by jose-rojas0** [werf/werf#5336] [General]
Hello,

I am intermittently getting failures like this while building my application using werf
```
Error: phase build on image my-app-image stage dockerfile handler failed: failed to build image for stage dockerfile with digest bec560428fe6e79516d8f7afc782007b3c2849041423e9af91bd7915: error building dockerfile with local-docker-server: Status: failed to export image: failed to create image: failed to get layer sha256:97b9c9d22887dcc64ed32292f28dab9703c9b72b743c1668f387c7f862c2efc3: layer does not exist, Code: 1
```

Usually retrying the `werf build` command succeeds

This seems possibly like the image is being cleaned up from docker at some point? However I have configured `WERF_DISABLE_AUTO_HOST_CLEANUP=true` in my environment. How can I prevent these failures?

**Comment by alexey-igrychev** [comment]
Hello, Jose, and sorry for the delay.

What werf version do you use? Do you face the problem after disabling werf auto host cleanup?
By itself, Docker cannot remove layers. You can use appropriate commands and automation, such as cron jobs, to do that — check this out.
